internal/sim/api.go

```diff
@@ -492,7 +504,44 @@ type registrar struct {
 	methods    map[string]reflect.Method              // non-Init exported methods, by name
 	regsByIntf map[reflect.Type]*codegen.Registration // registered components, by interface
 	fakes      map[reflect.Type]any                   // fakes, by component interface
-	generators map[string][]generator                 // generators, by method name
+	ops        []*op                                  // operations
+	opsByName  map[string]int                         // index into ops, by name
+}
+
+// newRegistrar returns a new registrar.
+func newRegistrar(t testing.TB, w reflect.Type, methods map[string]reflect.Method, regsByIntf map[reflect.Type]*codegen.Registration) *registrar {
+	ops := make([]*op, 0, len(methods))
+	opsByName := make(map[string]int, len(methods))
+	i := 0
+	for _, m := range methods {
+		ops = append(ops, &op{
+			name:       m.Name,
+			arity:      m.Type.NumIn() - 2,
+			generators: []generator{},
+			f:          m.Func,
+		})
+		opsByName[m.Name] = i
+		i++
+	}
+	return &registrar{
+		t:          t,
+		w:          w,
+		methods:    methods,
+		regsByIntf: regsByIntf,
+		fakes:      map[reflect.Type]any{},
+		ops:        ops,
+		opsByName:  opsByName,
+	}
+}
+
+// reset resets a registrar.
+func (r *registrar) reset() {
```

Comment:
It would be nice to make it clear which fields are immutable and which are modified.

Reply:
Added comments to the `registrar` struct to clarify this.